### ISSUE TITLE
Make new file(s), when the picture is rotated or flip, do not overwrite original file!

### ISSUE THREAD

**Issue by faramonius** [deckerst/aves#512]
**Is your feature request related to a problem? Please describe.**
When I tap on the menu item or the image rotation button, the program instantly overwrites the original file, and overwrites it without asking for any confirmation from the user.

**Describe the solution you'd like**
Rotating and flipping pictures should not change the original file! Moreover, I can accidentally press the button without wanting it. I propose to use rotation without autosaving the file, but using only RAM. In case it is not enough, use temporary files. I propose to add a separate "Save Changes" button, and always save a copy of the original file, using a prompt for the user to enter a new file name or to rename the file, or add an auto-rename mode.

**Comment by deckerst** [comment]
> Rotating and flipping pictures should not change the original file

I disagree. These changes are lossless and reversible, as they only modify the metadata.

**Comment by dataprolet** [comment]
This also affects editing pictures in a way that is neither lossless nor reversible for example cropping a picture. Please add a feature that the editor is asking whether you would like to overwrite the existing picture or rename it.

**Comment by deckerst** [comment]
Aves does not include an editor. When you use the "edit" action in Aves, it uses the default editor app on your device (and asks if there is no default).

**Comment by zilexa** [comment]
> > Rotating and flipping pictures should not change the original file
> 
> I disagree. These changes are lossless and reversible, as they only modify the metadata.

🙏🏽
Came here just to double check 'how' Aves rotates and happy to read it doesn't actually create a new re-oriented and re-encoded image, but only updates it's metadata! This means the original datetime values (date created, date digitized) are preserved? I thought I read somewhere its very hard to preserve date created on Android.

**Comment by deckerst** [comment]
> This means the original datetime values (date created, date digitized) are preserved? I thought I read somewhere its very hard to preserve date created on Android.

Depends what date you're talking about. Existing metadata dates are preserved. File dates may change.
Cf https://github.com/deckerst/aves/wiki/FAQ#whats-in-a-date

**Comment by zilexa** [comment]
Man, I wish Ente would hire you to fix their Ente Photos app.. its probably the only Google Photos alternative out there.. but their app totally sucks when it comes to rotating etc. They create a copy. Then the user will think they should delete the original. And now they lost all metadata. And ofc the file dates are of today. And more issues like that. It doesn't. End. 
They need someone like you. Because you even know how to explain this stuff in layman's terms! Thanks!
